syntax = "proto3";

package inference;

service InferenceService {
  rpc Detect (DetectRequest) returns (DetectResponse);
}

message DetectRequest {
  bytes image_data = 1;
  string camera_id = 2;
}

message DetectResponse {
  repeated Detection detections = 1;
}

message Detection {
  float x1 = 1;
  float y1 = 2;
  float x2 = 3;
  float y2 = 4;
  float confidence = 5;
  int32 class_id = 6;
  string class_name = 7;
}
